[24.0] Extend regex groups in stdio regex matches

bernt-matthias opened the pull request:
Allow to extend groups in the matching regex in the description.

Wondering where to description is shown? Is this broken? The screenshot shows the job info page which shows that the `Job Messages` are shown as dict and the job details in the history show `An error has occurred with this dataset`.

![Screenshot from 2023-11-13 15-30-29](https://github.com/galaxyproject/galaxy/assets/25689525/5093d563-5401-450c-89d0-453fdced23b5)



## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

bernt-matthias committed:
extend regex groups in stdio regex matchs

idea came up here https://github.com/galaxyproject/tools-iuc/issues/5629#issuecomment-1807050561

bernt-matthias committed:
temp change to test tool

nsoranzo commented on lib/galaxy/tool_util/output_checker.py:
```diff
@@ -176,8 +176,9 @@ def __regex_err_msg(match, stream, regex):
 
     # If there's a description for the regular expression, then use it.
     # Otherwise, we'll take the first 256 characters of the match.
+    print(f"{regex.desc=}")
```
Debug?

```suggestion
```

nsoranzo commented on lib/galaxy/tool_util/xsd/galaxy.xsd:
```diff
@@ -6757,6 +6757,27 @@ stderr are checked. If ``Branch A`` is at the beginning of stdout or stderr, the
 a warning will be registered and the source that contained ``Branch A`` will be
 prepended with the warning ``Warning: Branch A was taken in execution``.
 
+Since Galaxy 24.0 groups defines in the regular expression are expanded in the
```
```suggestion
Since Galaxy 24.0 groups defined in the regular expression are expanded in the
```

bernt-matthias committed:
Remove debug message

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

bernt-matthias committed:
Fix typo

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

nsoranzo commented on lib/galaxy/tool_util/xsd/galaxy.xsd:
```diff
@@ -6757,6 +6757,27 @@ stderr are checked. If ``Branch A`` is at the beginning of stdout or stderr, the
+i.e. the content of the first group.
+The second regular expression defines a named group ``error_message``
+which then replaces the corresponding placeholder ``\g<error_message>`` in the
+description. Notethe quoting of the ``<`` and ``>`` characters in XML.
```
```suggestion
description. Note the quoting of the ``<`` and ``>`` characters in XML.
```

bernt-matthias committed:
Fix typo

Co-authored-by: Nicola Soranzo <nicola.soranzo@gmail.com>

bernt-matthias committed:
fix tests

bernt-matthias committed:
make regex string

bernt-matthias committed:
remove unused variable job_id_tag

was used in the past for log messages which are now gone

bernt-matthias committed:
formating fixes

bernt-matthias committed:
fix test

bernt-matthias committed:
fix api test

bernt-matthias committed:
make all job messages a dict

fixes https://github.com/galaxyproject/galaxy/issues/17023

I'm unsure about storing job messages in such an
unstructured way...

bernt-matthias committed:
add types

mvdbeek commented on lib/galaxy/metadata/set_metadata.py:
```diff
@@ -340,7 +348,15 @@ def set_meta(new_dataset_instance, file_dict):
+                {
+                    "type": "max_discovered_files",
+                    "desc": str(e),
+                    "code_desc": str(e),
```
That's meant for exit code descriptions.
```suggestion
                    "code_desc": None,
```
should be fine

mvdbeek commented on lib/galaxy/metadata/set_metadata.py:
```diff
@@ -340,7 +348,15 @@ def set_meta(new_dataset_instance, file_dict):
+            job_messages.append(
+                {
+                    "type": "max_discovered_files",
+                    "desc": str(e),
```
```suggestion
                    "desc": "Too many output files were discovered for a single job.",
```

bernt-matthias committed:
Improve job_message for max discovered files

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>

bernt-matthias committed:
linter fixes

bernt-matthias committed:
fix test exit code default

mvdbeek commented on lib/galaxy/metadata/set_metadata.py:
```diff
@@ -338,9 +346,17 @@ def set_meta(new_dataset_instance, file_dict):
             )
             collect_dynamic_outputs(job_context, output_collections)
-        except MaxDiscoveredFilesExceededError as e:
+        except MaxDiscoveredFilesExceededError:
```
```suggestion
        except MaxDiscoveredFilesExceededError as e:
```

mvdbeek commented on lib/galaxy/metadata/set_metadata.py:
```diff
@@ -338,9 +346,17 @@ def set_meta(new_dataset_instance, file_dict):
+            job_messages.append(
+                {
+                    "type": "max_discovered_files",
+                    "desc": "Too many output files were discovered for a single job.",
```
```suggestion
                    "desc": str(e),
```
I'm sorry, that's actually a better message since it includes the limit.

bernt-matthias committed:
Apply suggestions from code review

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>

bernt-matthias committed:
remove unneded classes

bernt-matthias committed:
black reformatting

bernt-matthias committed:
fix import order

bernt-matthias committed:
full fix for import sorting ..

lets see if this conflicts with black

mvdbeek committed:
Fix up integration test for max discovered files

bgruening commented on lib/galaxy/tool_util/xsd/galaxy.xsd:
```diff
@@ -6757,6 +6757,27 @@ stderr are checked. If ``Branch A`` is at the beginning of stdout or stderr, the
 a warning will be registered and the source that contained ``Branch A`` will be
 prepended with the warning ``Warning: Branch A was taken in execution``.
 
+Since Galaxy 24.0 groups defined in the regular expression are expanded in the
```
23.2 or 24.0?

bernt-matthias replied:
```suggestion
Since Galaxy 23.2 groups defined in the regular expression are expanded in the
```

bernt-matthias committed:
Update lib/galaxy/tool_util/xsd/galaxy.xsd

hujambo-dunia committed:
Fix, basic - preformatted html tag

hujambo-dunia committed:
Fix, moderate - bulletted list

hujambo-dunia committed:
Add check if output is not an array of objects

bernt-matthias committed:
make the max discovered files error a "proper" job error message

jmchilton committed:
Links to invocations. (by John Chilton via Michelle Savage Vue-component upgrade)
Upgraded component to Vue 3. Extra space removed.
Remove index enumeration and indentation; separate multiple errors with border line instead. Prevent metadata from printing when there is no value.
Rebased and edited/removed accidental 'merge conflict markers' in file

bernt-matthias commented on lib/galaxy/tool_util/xsd/galaxy.xsd:
```diff
@@ -7009,6 +7009,27 @@ stderr are checked. If ``Branch A`` is at the beginning of stdout or stderr, the
 a warning will be registered and the source that contained ``Branch A`` will be
 prepended with the warning ``Warning: Branch A was taken in execution``.
 
+Since Galaxy 23.2 groups defined in the regular expression are expanded in the
```
Lets also do this once the history is clean:

```suggestion
Since Galaxy 24.0 groups defined in the regular expression are expanded in the
```

jdavcs committed:
Update lib/galaxy/tool_util/xsd/galaxy.xsd

Co-authored-by: M Bernt <m.bernt@ufz.de>

jdavcs reviewed (approved):
Looks good to me. Waiting for the tests before merging (there may will be failures that are unrelated). Thanks again for addressing the comments and handling the rebase, @hujambo-dunia!

jdavcs committed:
Fix client linting
